### PR TITLE
[IMP] base: improve performances of commercial fields

### PR DESCRIPTION
**Pull request by benoit-phytocontrol**
This PR improve performances by reducing the number of SQL queries when writing a res.partner's commercial field.
    
Test case: write property_account_position_id on a partner with 10 children (tested with the code above)

Before this PR: 120 SQL queries
After this PR: 12 SQL queries

I tested on all versions 14.0 to 17.0:

  | Odoo 14.0 | Odoo 15.0 | Odoo 16.0 | Odoo 17.0
-- | -- | -- | -- | --
queries before this commit | 252 | 263 | 142 | 120
queries after this commit | 22 | 22 | 12 | 12

The difference is even greater if specific modules have added additional commercial fields

### CODE REVIEW

**Comment by Julien00859** [comment]
@robodoo delegate @kmagusiak

File: odoo/addons/base/models/res_partner.py [hunk 1]
```diff
@@ -607,15 +607,16 @@ def _commercial_sync_from_company(self):
             self.write(sync_vals)
             self._commercial_sync_to_children()
 
-    def _commercial_sync_to_children(self):
+    def _commercial_sync_to_children(self, fields_to_sync=None):
```
[the remark this answers is not in the document]
Comment (benoit-phytocontrol) [reply odoo/addons/base/models/res_partner.py:610]: What do you recommend ?
- Create a new function instead and keep _commercial_sync_to_children() with current signature for compatibility
- Edit the PR for merge in master

[the remark this answers is not in the document]
Comment (kmagusiak) [reply odoo/addons/base/models/res_partner.py:610]: Just create a PR for master.

[the remark this answers is not in the document]
Comment (benoit-phytocontrol) [reply odoo/addons/base/models/res_partner.py:610]: New PR : https://github.com/odoo/odoo/pull/173540

**Comment by Julien00859** [comment]
Superseded by: https://github.com/odoo/odoo/pull/173540

**Pull request closed without merging**